### ISSUE TITLE
Always return a route

### ISSUE THREAD

**Issue by HarelM** [perliedman/geojson-path-finder#86]
When the start or end point are not close enough to the graph there is no path results, or at least this is what I've seen in the demo page, I have yet to take this library for a spin.
It would be great if there was an option to return a route from the closest point on the graph to the start and end, this way, if you have a valid graph you will always have a path results.
Am I missing something in the docs?
Generally speaking, this is how graphhopper works and I think it's a good feature to always try and return a route.

**Comment by HarelM** [comment]
This is documented basically:
>Where `start` and `finish` are two GeoJSON point features. Note that both points have to be vertices in the routing network; if they are not, no route will be found.

So I would like to reduce this requirement in general.

**Comment by HarelM** [comment]
Yes, I just implemented that, but I think that it should be part of this routing library.
I'll add the code at the end if anyone is interested.
Here's another scenario that is currently not supported but I think should be:
You have a starting point which is not on a vertex, and you want the library to find the route.
If you simply select the closest vertex, or the closest vertex on the closest line you might get an incorrect results.
Here's an example of selecting the closest vertex:
<img width="327" alt="image" src="https://user-images.githubusercontent.com/3269297/227508287-b827058c-8057-4f53-846b-c3a6288239e1.png">
As can be seen in the image, the start and end points are on the path, but they are not vertices, so the path finder start a bit after the first point and finish a bit after the last point... This is not a great UX. I'm sure I'm not the first to stumble across this.
We are using graphhopper on the server side and I want to introduce this as a back when there's no connection.
Graphhopper does exactly this - finds the closest projected point on the closest line and does the routing using it.
I think this is the expected UX from this library.
I'll be happy to help out contributing to this code base.

Here's the code I wrote to fix both issues, it uses `turf` obviously and is performed before the graph is created.
This is less than ideal if I would like to cache the graph...
```ts
    /**
     * This method will insert a single point to the closest line in the collection.
     * The point that will be added will be on the closest segment of the closest line.
     * @param latlng the point to add a projected version to the closest line
     * @param collection the collection of all the lines to test against
     * @returns the projected point
     */
    public static insertProjectedPointToClosestLine(latlng: LatLngAlt, collection: FeatureCollection<GeoJSON.LineString>) {
        let closetLine = null;
        let nearestPoint = null;
        let minimalDistance = Infinity;
        let coordinates = SpatialService.toCoordinate(latlng);
        for (let feature of collection.features) {
            let point = nearestPointOnLine(feature, coordinates, )
            if (point.properties.dist < minimalDistance) {
                minimalDistance = point.properties.dist;
                closetLine = feature;
                nearestPoint = point;
            }
        }
        closetLine.geometry.coordinates.splice(nearestPoint.properties.index + 1, 0, nearestPoint.geometry.coordinates);
        return nearestPoint.geometry.coordinates;
    }```

**Comment by perliedman** [comment]
I'm going to agree with @pbvahlst here and say this is not the scope of this library: it might make sense for something like GraphHopper where the routing network is generelly dense and have large geographic cover, but this is often not the case for this library. Just always snapping to the closest node doesn't make sense in a lot of cases and I feel the heuristics involved to do this depends on the application.
